feat(serde): kafka format (#3065)
confluentinc/ksql: docs/developer-guide/serialization.rst

For more information, see :ref:`ksql_single_field_wrapping`.

Field Name Case Sensitivity
---------------------------

The format is case-insensitive when matching a KSQL field name with an Avro record's field name.
The first case-insensitive match is used.

.. _kafka_format

-----
KAFKA
-----

The ``KAFKA`` format supports``INT``, ``BIGINT``, ``DOUBLE`` and ``STRING`` primitives that have
been serialized using Kafka's standard set of serializers.

The format is designed primarily to support primitive message keys. It can be used as a value format,

> though certain operations are not yet supported when this is the case.

Reviewer: ```suggestion
though certain operations aren't supported when this is the case.
```

> Unlike some other formats, this format does not perform any type coercion. It is therefore important

Reviewer: ```suggestion
Unlike some other formats, the ``KAFKA`` format does not perform any type coercion, so it's important
```

to correctly match the field type to the underlying serialized form to avoid deserialization errors.

+-------------------------+--------------------------------------------------------------------------------------------+
| KSQL Field Type         | Kafka Type                                                                                 |
+=========================+============================================================================================+

> | INT / INTEGER           | A 32-bit signed integer, as has been serialized using the Java                             |

Reviewer: Maybe: "A 32-bit signed integer, that was serialized by using the Java..."
Or: "A 32-bit signed integer serialized by using the Java..."

| BIGINT                  | A 64-bit signed integer, as has been serialized using the Java                             |
|                         | ``org.apache.kafka.common.serialization.LongSerializer``, or equivalent.                   |
+-------------------------+--------------------------------------------------------------------------------------------+
| DOUBLE                  | A 64-bit floating point number, as has been serialized using the Java                      |
|                         | ``org.apache.kafka.common.serialization.DoubleSerializer``, or equivalent.                 |
+-------------------------+--------------------------------------------------------------------------------------------+
| STRING / VARCHAR        | A UTF-8 encoded text string, as has been serialized using the Java                         |
|                         | ``org.apache.kafka.common.serialization.StringSerializer``, or equivalent.                 |
+-------------------------+--------------------------------------------------------------------------------------------+

> As the format only supports primitive types it can only be used where the schema contains a single field.

Reviewer: ```suggestion
Because the format supports only primitive types, you can only use it when the schema contains a single field.
```

> For example, if your Kafka messages have a ``long`` key you can make them available to KSQL a statement

Reviewer: ```suggestion
For example, if your Kafka messages have a ``long`` key, you can make them available to KSQL a statement
```
